### PR TITLE
Tweak verification UI.

### PR DESCRIPTION
**Pull request by charlesmchen**
* Add “tap for options” to verification banner copy.
* Add close button to banners.
* Always use shield icon for verification row in conversation settings view.
* Add action sheet to verification banner.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 1]
```diff
@@ -675,6 +675,19 @@ - (void)setUserHasScrolled:(BOOL)userHasScrolled
...
+- (NSArray<NSString *> *)noLongerVerifiedRecipientIds
+{
+    NSMutableArray<NSString *> *result = [NSMutableArray new];
+    for (NSString *recipientId in self.thread.recipientIdentifiers) {
+        if ([[OWSIdentityManager sharedManager] verificationStateForRecipientId:recipientId]
+            == OWSVerificationStateNoLongerVerified) {
+            [result addObject:recipientId];
+        }
+    }
+    return result;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: nit: copy to return immutable.

File: Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m [hunk 2]
```diff
@@ -240,16 +240,12 @@ - (void)updateTableContents
...
-        BOOL isVerified = [[OWSIdentityManager sharedManager] verificationStateForRecipientId:recipientId]
-            == OWSVerificationStateVerified;
-
         [firstSection addItem:[OWSTableItem itemWithCustomCellBlock:^{
             return [weakSelf
                 disclosureCellWithName:
                     NSLocalizedString(@"VERIFY_PRIVACY",
                         @"Label for button or row which allows users to verify the safety number of another user.")
-                              iconName:(isVerified ? @"table_ic_verify" : @"table_ic_not_verified")];
+                              iconName:@"table_ic_not_verified"];
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m]: ah, good, so per my earlier comment, looks like we're not changing this icon depending on verification state. 

nit: the icon filename is weird.

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 3]
```diff
@@ -814,7 +832,57 @@ - (void)blockBannerViewWasTapped:(UIGestureRecognizer *)sender
...
 {
     if (sender.state == UIGestureRecognizerStateRecognized) {
-        [self showConversationSettingsAndShowVerification:YES];
+        UIAlertController *actionSheetController =
+            [UIAlertController alertControllerWithTitle:nil
+                                                message:nil
+                                         preferredStyle:UIAlertControllerStyleActionSheet];
+
+        __weak MessagesViewController *weakSelf = self;
+        UIAlertAction *unblockAction = [UIAlertAction
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: a little confused, but I think it's just a failure to update variable names after copy/paste.

This should be something like `verifyPrivacyAction` not `unblockAction` right?

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: Done.

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: Done.

File: Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m]: Yup.  The icon filename _is_ weird.

**Pull request merged**